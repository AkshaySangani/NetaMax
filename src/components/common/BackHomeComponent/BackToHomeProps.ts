export interface IBackToHomeProps {
  /**
   * The title that we want to set
   * @type {string}
   */
  title?: string;

  /**
   * The size of the text
   * @type {number}
   */
  size?: number;
  /**
   * The weight of the text
   * @type {number}
   */
  weight?: number;

  /**
   * Color of text and Chevron
   * @type {string}
   */
  color?: string;

  /**
   * The home route
   * @type {string}
   */
  homeRoute?: string;
}
