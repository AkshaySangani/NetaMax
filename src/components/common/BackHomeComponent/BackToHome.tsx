import React, { ReactElement } from "react";

import { FaChevronLeft } from "react-icons/fa";

import { useRouter } from "next/router";

import { Box, Text } from "@chakra-ui/react";

import { IBackToHomeProps } from "./BackToHomeProps";

/**
 * Back to home nav component
 * @param {IBackToHomeProps} props the component props
 * @returns {ReactElement} back to home component
 */
export const BackToHome = ({
  title,
  size,
  weight,
  homeRoute = "/",
  color = "black",
}: IBackToHomeProps): ReactElement => {
  const router = useRouter();

  /**
   * Back to home function
   * @param {string} homeRoute route to redirect
   * @returns {void} back to home
   **/
  const backToHome = (homeRoute: string) => {
    router.push(homeRoute);
  };

  return (
    <Box display="flex" alignItems="center" onClick={() => backToHome(homeRoute)} cursor="pointer">
      <FaChevronLeft color={color} />
      <Text
        ml="12px"
        fontSize={size ? `${size}px` : "16px"}
        fontWeight={weight ? `${weight}` : "600"}
        color={color}
      >
        {title ? title : "Volver a la tiendita"}
      </Text>
    </Box>
  );
};
