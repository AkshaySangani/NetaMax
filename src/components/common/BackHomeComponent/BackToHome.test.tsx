import { render, cleanup, screen, fireEvent } from "@testing-library/react";
import { BackToHome } from "./BackToHome";
import { IBackToHomeProps } from "./BackToHomeProps";
import mockRouter from "next-router-mock";
import singletonRouter from "next/router";

jest.mock("next/router", () => require("next-router-mock"));

describe("Render BackToHome component", () => {
  let mock: IBackToHomeProps;

  beforeEach(() => {
    mock = {
      title: "Volver a la tiendita",
      size: 16,
      weight: 600,
    };
  });

  afterEach(() => {
    cleanup();
    jest.clearAllMocks();
  });

  test("BackToHome should be render correctly", () => {
    const { container } = render(<BackToHome />);
    expect(container).toMatchSnapshot();
  });

  test("Show title correctly", () => {
    const { queryByText } = render(<BackToHome {...mock} />);
    expect(queryByText("Volver a la tiendita")).toBeTruthy();
  });
});

describe("Test backToHome function", () => {
  beforeEach(() => {
    mockRouter.setCurrentUrl("/");
  });

  it("Supports `push` method", () => {
    singletonRouter.push("/");
    expect(singletonRouter).toMatchObject({
      asPath: "/",
    });
  });

  it("Works with next/link", () => {
    render(<BackToHome />);
    fireEvent.click(screen.getByText("Volver a la tiendita"));
    expect(singletonRouter).toMatchObject({ asPath: "/" });
  });
});
