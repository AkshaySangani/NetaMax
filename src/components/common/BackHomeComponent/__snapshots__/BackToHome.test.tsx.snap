// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render BackToHome component BackToHome should be render correctly 1`] = `
<div>
  <div
    class="css-759u60"
  >
    <svg
      color="black"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      style="color: black;"
      viewBox="0 0 320 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M34.52 239.03L228.87 44.69c9.37-9.37 24.57-9.37 33.94 0l22.67 22.67c9.36 9.36 9.37 24.52.04 33.9L131.49 256l154.02 154.75c9.34 9.38 9.32 24.54-.04 33.9l-22.67 22.67c-9.37 9.37-24.57 9.37-33.94 0L34.52 272.97c-9.37-9.37-9.37-24.57 0-33.94z"
      />
    </svg>
    <p
      class="chakra-text css-1utd549"
    >
      Volver a la tiendita
    </p>
  </div>
</div>
`;
